### ISSUE TITLE
Feature request: Add support for saving all changes at once

### ISSUE THREAD

**Issue by hakanserce** [abrochard/emacs-todoist#10]
Support for editing all tasks directly and then finally pressing C-x C-s , for instance, to save all the changes, or more like synchronizing with todoist.com.

**Comment by abrochard** [comment]
That would be super cool but also pretty hard...

**Comment by ghost** [comment]
It would give freedom to edit the org buffer like usual and then sync everything. Something like in dired edit mode? 

- From the todoist-mode switch to "todoist-edit-mode" with C-c C-e
- The main commands will add an action property instead of executing f.ex 
   todoist-close-task would only add a property :ACTION: close
- New sections without PROPERTIES will be create like todoist-new
- Done editing with C-c C-c and perform a "todoist-sync" executing the action :ACTION: or creating new tasks.

**Comment by dmitrym0** [comment]
Could the todoist buffer be treated as a regular org buffer including bidirectional sync?

**Comment by abrochard** [comment]
It could and that would be a great feature, but it is pretty hard to implement. Two big issues I see already is
1. what happens if a change happens to the tasks outside of org-mode? like you check a task off on your phone but then edit it in org-mode. We would need frequent polling or some kind of conflict resolution strategy.
2. what if the user takes an action in org-mode that isn't supported by todoist? like messing up the formatting or erasing the task id tag? we can't sync that task anymore so what should we do with it? erase it? erase the change? leave it and potentially show a duplicate?

**Comment by dmitrym0** [comment]
You're right; this is not a trivial task. I'm just dreaming of a first class task tracking solution for mobile. Todoist, Omnifocus, RTM, etc they all have desktop-web-mobile sync so it's technically possible just not easy.

**Comment by Lillenne** [comment]
@hakanserce @dmitrym0 incremental, bidirectional sync of org with Todoist with command batching is implemented for most features of projects, subprojects, sections, tasks (including additional features such as effort and deadlines), subtasks, and comments in [org-todoist](https://github.com/Lillenne/org-todoist).
